misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -2494,13 +2494,12 @@ public TableConstructorExpressionNode transform(TableConstructorExpressionNode t
         KeySpecifierNode keySpecifier = formatNode(tableConstructorExpressionNode.keySpecifier().orElse(null), 1, 0);
         SeparatedNodeList<Node> rows = tableConstructorExpressionNode.rows();
         int rowTrailingWS = 0, rowTrailingNL = 0;
-        if (rows.size() > 1) {
+        if (rows.size() > 1 || (rows.size() == 1 && shouldExpand(rows.get(0)))) {
```

Comment:
Wouldnt it be a better design to move this logic into the util itself? (So that we keep all the related logic in `shouldExpand(TableConstructorExpressionNode node)`

Reply:
Addressed this comment.